server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -535,13 +567,33 @@ private void executeLocalSearch(Task task, SearchTimeProvider timeProvider, Sear
                 remoteShardIterators.add(shardIterator);
             }
         }
-        return (clusterAlias, nodeId) -> {
-                Map<String, DiscoveryNode> clusterNodes = clusterToNode.get(clusterAlias);
-                if (clusterNodes == null) {
-                    throw new IllegalArgumentException("unknown remote cluster: " + clusterAlias);
+        return remoteShardIterators;
+    }
+
+    static List<SearchShardIterator> getRemoteShardsIteratorFromPointInTime(Map<String, ClusterSearchShardsResponse> searchShardsResponses,
+                                                                            SearchContextId searchContextId,
+                                                                            TimeValue searchContextKeepAlive,
+                                                                            Map<String, OriginalIndices> remoteClusterIndices) {
+        final List<SearchShardIterator> remoteShardIterators = new ArrayList<>();
+        for (Map.Entry<String, ClusterSearchShardsResponse> entry : searchShardsResponses.entrySet()) {
+            for (ClusterSearchShardsGroup group : entry.getValue().getGroups()) {
+                final ShardId shardId = group.getShardId();
+                final String clusterAlias = entry.getKey();
+                final SearchContextIdForNode perNode = searchContextId.shards().get(shardId);
+                assert clusterAlias.equals(perNode.getClusterAlias()) : clusterAlias + " != " + perNode.getClusterAlias();
+                final List<String> nodeIds = new ArrayList<>();
+                nodeIds.add(perNode.getNode()); // always search with the matching node first
+                for (ShardRouting shard : group.getShards()) {
+                    if (perNode.getNode().equals(shard.currentNodeId()) == false) {
+                        nodeIds.add(shard.currentNodeId());
```

Comment:
While this is something that we'll need in the future I don't think we can afford adding replicas other than the original one now. There's no chance that the query succeeds there so that's confusing for users (and could faild some tests) if we return a search context missing exception every time there's an exception on the matching node.

Reply:
++. Let's reintroduce this when we implement the retry. See: https://github.com/elastic/elasticsearch/pull/61827/commits/0a99df6e9a3cf1834b19e23bc9f2076358ab94ed